src/usdb_syncer/gui/mw.py

```diff
@@ -409,6 +415,60 @@ def _rate_in_usdb(self, stars: int) -> None:
         else:
             logger.debug("Not rating song: no song selected.")
 
+    def _submit_to_usdb(self) -> None:
+        """Submit song changes to USDB.
+
+        Normal users may only submit one song at a time (the currenty selected song).
+        Moderators may submit all selected local songs.
+        """
+
+        if (user := SessionManager.get_user()) and user.role in {
+            UserRole.ADMIN,
+            UserRole.MODERATOR,
+        }:
+            candidates: Iterable[UsdbSong | None] = list(self.table.selected_songs())
+        else:
+            candidates = [self.table.current_song()]
+
+        local_songs = [s for s in candidates if s and s.is_local()]
+        num_songs = len(local_songs)
+
+        if not local_songs:
+            QMessageBox.information(
+                self, "No Local Songs", "No local songs available for submission."
+            )
+            return
+
+        if num_songs == 1:
+            song = local_songs[0]
+            message = f"Submit changes for: {song.artist} - {song.title}?"
+        else:
+            message = f"Submit changes for {num_songs} local songs?"
+
+        reply = QMessageBox.question(
+            self,
+            "Confirm Submission",
+            message,
+            QMessageBox.StandardButton.Yes | QMessageBox.StandardButton.No,
+            QMessageBox.StandardButton.No,
+        )
+
+        if reply != QMessageBox.StandardButton.Yes:
+            return
+
+        def task() -> None:
+            for song in local_songs:
+                submit_local_changes(song)
+
+        def on_done(result: progress.Result) -> None:
+            pass
+
+        run_with_progress(
+            f"Submitting {num_songs} song{'s' if num_songs != 1 else ''}…",
+            task,
+            on_done,
```

Comment:
This silently discards any exception.

Reply:
So what can be done about it? Run it without progress? For the normal user, it is limited to a single song anyways, so I guess this is only relevant for batch uploads by mods/admins.